### PR TITLE
fix: [PIE-9798]: Added setting to enable matrix labels by name instead of keys

### PR DESCRIPTION
**Pull request by ankittiwari-harness**
### Summary

Added the pipeline settings for enabling a true / false flag for allowing accounts to change the way the matrix are labelled. 

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by sahilhindwani-harness — APPROVED** [review]

File: src/modules/23-default-settings/strings/strings.en.yaml [hunk 1]
```diff
@@ -13,3 +13,4 @@ stageTimeout: Stage Timeout
 stepTimeout: Step Timeout
 concurrentActivePipelineExecutions: Concurrent Active Pipeline Executions
 allowUserToMarkStepAsFailedExplicitly: Allow users to mark a running Step as failure
+enableMatrixFieldNames: Enable matrix Labels By Name
```
Comment (rameshx) [inline src/modules/23-default-settings/strings/strings.en.yaml:16]: Every word is capitalized except `matrix`, can this be `Enable matrix labels by name`?

Comment (chetanvekariya-harness) [reply src/modules/23-default-settings/strings/strings.en.yaml:16]: +1

Comment (ankittiwari-harness) [reply src/modules/23-default-settings/strings/strings.en.yaml]: Done in Rev

Comment (chetanvekariya-harness) [reply src/modules/23-default-settings/strings/strings.en.yaml]: +1

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by rameshx — DISMISSED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by Eric761 — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**